synapse/types.py

```diff
@@ -19,7 +19,7 @@
 
 
 Requester = namedtuple("Requester",
-                       ["user", "access_token_id", "is_guest", "device_id"])
+                       ["user", "access_token_id", "is_guest", "device_id", "as_user"])
```

Comment:
I'm not sure I like the name, given that its an `ApplicationService` rather than some AS user type. I would have said `associated_application_service` if it wasn't so long. Or maybe `sending_as`?